distributed/comm/tcp.py

```diff
@@ -190,18 +190,12 @@ def peer_address(self):
             lengths = await stream.read_bytes(8 * n_frames)
             lengths = struct.unpack("Q" * n_frames, lengths)
 
-            frames = []
-            for length in lengths:
-                if length:
-                    if self._iostream_has_read_into:
-                        frame = bytearray(length)
-                        n = await stream.read_into(frame)
-                        assert n == length, (n, length)
-                    else:
-                        frame = await stream.read_bytes(length)
-                else:
-                    frame = b""
-                frames.append(frame)
+            frames = [bytearray(each_length) for each_length in lengths]
+            recv_frames = [each_frame for each_frame in frames if len(each_frame) > 0]
+            for each_frame in recv_frames:
+                each_length = len(each_frame)
+                n = await stream.read_into(each_frame)
+                assert n == each_length, (n, each_length)
```

Comment:
It looks like this change has a semantic difference in that previously we used to include empty frames but now we don't.  Is this correct?

Is there a specific reason for this change?  Does it affect performance in some way, or is it strictly cosmetic?  If it's strictly cosmetic, and if there is a change to semantics (however minor) I'm tempted to avoid the change just because I wouldn't be surprised if it has some unforeseen effect.

Reply:
That's not true actually. We [excluded them before]( https://github.com/dask/distributed/blob/e928cc0090015a7e07c18d2d64255d33a849224a/distributed/comm/tcp.py#L195 ) as well. It's just now done outside of the `for`-loop. 🙂

The idea was to more-or-less pass control to Tornado or UCX-Py completely while receiving frames without all of the boilerplate in Dask between each receive. Tried `asyncio.gather` to push this even further, but Tornado doesn't seem to like that.

In any event I thought it would be a nice thing to do the same thing we are doing in UCX for TCP. At the end of the day, I don't have strong feelings about it (especially if there is push back). 🤷‍♂

Reply:
Any other thoughts here, @mrocklin? 🙂